graylog2-web-interface/src/components/indices/IndicesMaintenanceDropdown.tsx

```diff
@@ -54,6 +58,9 @@ const IndicesMaintenanceDropdown = ({ indexSet, indexSetId }: Props) => {
       <DropdownButton bsStyle="info" title="Maintenance" id="indices-maintenance-actions" pullRight>
         <MenuItem eventKey="1" onClick={onRecalculateIndexRange}>Recalculate index ranges</MenuItem>
         {cycleButton}
+        {indexSet?.data_tiering_status?.has_failed_snapshot && dataTieringPlugin && (
+          <dataTieringPlugin.DeleteFailedSnapshotMenuItem eventKey="3" indexSetId={indexSetId} />
+        )}
```

Comment:
Ideally, this would also not live in the server, but for the sake of moving this forward let's leave it like this.